File: icando-ae-fe/apps/assessment-engine/src/interfaces/question.ts
import { Competency } from "./competency.ts";

export interface Choice {
  id: number;
  text: string;
}

export interface Question {
  id: string;
  choices: Choice[];
  text: string;
  answerId: number;
  quizId: string;
  competencies: Competency[];
}

File: icando-ae-api/internal/model/dto/question.go
package dto

import (
	"github.com/google/uuid"
)

type GetQuestionFilter struct {
	ID     uuid.UUID
	QuizID uuid.UUID
}

type QuestionDto struct {
	Text         string              `json:"text"`
	Choices      []QuestionChoiceDto `json:"choices"`
	AnswerID     int                 `json:"answer_id"`
	Competencies []uuid.UUID         `json:"competencies"`
}

type QuestionChoiceDto struct {
	ID   int    `json:"id"`
	Text string `json:"text"`
}
